go/vt/vttablet/tabletmanager/vreplication/vplayer.go

```diff
@@ -403,6 +404,45 @@ func (vp *vplayer) applyEvent(ctx context.Context, event *binlogdatapb.VEvent, m
 				return io.EOF
 			}
 		}
+	case binlogdatapb.VEventType_JOURNAL:
+		switch event.Journal.MigrationType {
+		case binlogdatapb.MigrationType_SHARDS:
+			// no-op
```

Comment:
Why is this a no-op?

Reply:
This is only the validation. For a reshard, all migrated. So, no need to validate. For tables, we have to make sure all tables in the stream were migrated. I've added more comments.